codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/config/ServiceConfigGenerator.kt

```diff
@@ -81,12 +89,72 @@ sealed class ServiceConfig(name: String) : Section(name) {
      * A section for extra functionality that needs to be defined with the config module
      */
     object Extras : ServiceConfig("Extras")
+
+    /**
+     * The set default value of a field for use in tests, e.g `${configBuilderRef}.set_credentials(Credentials::for_tests())`
+     */
+    data class DefaultForTests(val configBuilderRef: String) : ServiceConfig("DefaultForTests")
+}
+
+data class ConfigParam(val name: String, val type: Symbol, val setterDocs: Writable?, val getterDocs: Writable? = null)
+
+/**
+ * Config customization for a standard config param:
+ * 1. `pub(crate)` field
+ * 2. pub(crate) accessor
```

Comment:
It feels weird to backtick one but not the other
```suggestion
 * 1. `pub(crate)` field
 * 2. `pub(crate)` accessor
```